baybe/searchspace/discrete.py

```diff
@@ -697,27 +728,34 @@ def _apply_polars_constraint_filter(
     return ldf
 
 
-def _apply_constraint_filter(
-    df: pd.DataFrame, constraints: Collection[DiscreteConstraint]
-):
-    """Remove discrete search space entries inplace based on constraints.
+def parameter_cartesian_prod_polars(parameters: Iterable[Parameter]) -> pl.LazyFrame:
+    """Create the Cartesian product of all parameter values.
+
+    Ignores continuous parameters.
 
     Args:
-        df: The data in experimental representation to be modified inplace.
-        constraints: List of discrete constraints.
+        parameters: List of parameter objects.
 
+    Returns:
+        A lazy dataframe containing all possible discrete parameter value combinations.
     """
-    # Reorder the constraints according to their execution order
-    constraints = sorted(
-        constraints,
-        key=lambda x: DISCRETE_CONSTRAINTS_FILTERING_ORDER.index(x.__class__),
-    )
+    discrete_parameters = [p for p in parameters if isinstance(p, DiscreteParameter)]
+    if not discrete_parameters:
+        return pl.LazyFrame()
 
-    # Remove entries that violate parameter constraints:
-    for constraint in (c for c in constraints if c.eval_during_creation):
-        idxs = constraint.get_invalid(df)
-        df.drop(index=idxs, inplace=True)
-    df.reset_index(inplace=True, drop=True)
+    # Convert each parameter to a lazy dataframe for cross-join operation
+    param_frames = [pl.LazyFrame({p.name: p.values}) for p in discrete_parameters]
+
+    # Handling edge cases
+    if len(param_frames) == 1:
+        return param_frames[0]
+
+    # Cross-join parameters
+    res = param_frames[0]
+    for frame in param_frames[1:]:
+        res = res.join(frame, how="cross", force_parallel=True)
+
+    return res
 
 
 def parameter_cartesian_prod_to_df(
```

Comment:
please rename to make clear this is the pandas version similar to the versions of constraint applicators